x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/convert/ToLong.java

```diff
@@ -81,16 +82,16 @@ static long fromBoolean(boolean bool) {
         return bool ? 1L : 0L;
     }
 
-    @ConvertEvaluator(extraName = "FromString", warnExceptions = { NumberFormatException.class })
+    @ConvertEvaluator(extraName = "FromString", warnExceptions = { InvalidArgumentException.class })
     static long fromKeyword(BytesRef in) {
         String asString = in.utf8ToString();
         try {
-            return Long.parseLong(asString);
-        } catch (NumberFormatException nfe) {
+            return stringToLong(asString);
+        } catch (InvalidArgumentException iae) {
             try {
-                return fromDouble(Double.parseDouble(asString));
+                return fromDouble(stringToDouble(asString));
             } catch (Exception e) {
-                throw nfe;
+                throw iae;
```

Comment:
Potential optimization is to create a dedicated method that does `stringToSmallestInteger` and look into custom parsing that avoids having to catch exceptions before falling back. Can be a separate PR but in that case please raise an issue for it.

Reply:
I'll make the change in this PR.